MatrixKit/Models/Room/MXKRoomBubbleCellData.m

```diff
@@ -587,6 +587,21 @@ - (BOOL)hasAttributedTextMessage
     return hasAttributedTextMessage;
 }
 
+- (BOOL)hasLink
+{
+    @synchronized (bubbleComponents) {
+        for (MXKRoomBubbleComponent *component in bubbleComponents)
```

Comment:
In Element we break up the bubbleComponents so each cell data object has only 1 link and it is always from the last component in the data: 
An optimisation here could be to iterate through the components backwards or possibly to look directly at the last component (although this behaviour would be poor for other clients using the framework).